include/stl2/detail/iterator/move_iterator.hpp

```diff
@@ -90,6 +91,33 @@ STL2_OPEN_NAMESPACE {
 				static_cast<void>(++current_)
 			)
 
+			// Experimental support for move_iterator post-increment
+			using __post_inc_t = decltype(current_++);
+			Readable{R}
+			struct __proxy {
+				R __tmp;
+				decltype(auto) operator*()
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(__tmp)
+				)
+				friend decltype(auto) iter_move(const __proxy& that)
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(that.__tmp)
+				)
+			};
+
+			STL2_CONSTEXPR_EXT cursor post_increment()
+			noexcept(noexcept(cursor{cursor{__stl2::exchange(current_, __stl2::next(current_))}}))
+			requires Same<__post_inc_t, I>() {
+				return cursor{__stl2::exchange(current_, __stl2::next(current_))};
```

Comment:
> This concerns me: can we prove that `current_` is valid after `next(current_)`, and that the value of `*current_` must be unchanged?

My thinking is that even if `I`'s postfix increment operator is broken wrt its semantics, `move_iterator<I>`'s postfix increment operator is no _more_ broken than `I`'s.

> Ouch, we can't derive this for `InputIterator` now _with_ post increment: there's no semantic requirement that `*i++` is equal to the previous value of `*i`.

So, shall we add normative wording that _if_ the type of `i++` satisfies `Readable` that the expression `*i++` should have the same value as the expression `*i`? I can include that in my paper to nuke the Readability requirement on `i++`.